android/src/main/java/ee/forgr/capacitor_updater/CapacitorUpdater.java

```diff
@@ -440,62 +451,37 @@ public VersionInfo getVersionInfoByName(final String version) {
     }
 
     private void removeVersionInfo(final String version) {
-        this.setVersionDownloadedTimestamp(version, null);
-        this.setVersionName(version, null);
-        this.setVersionStatus(version, null);
-    }
-
-    private String getVersionDownloadedTimestamp(final String version) {
-        return this.prefs.getString(version + DOWNLOADED_SUFFIX, "");
+        this.saveVersionInfo(version, null);
     }
 
-    private void setVersionDownloadedTimestamp(final String version, final Date time) {
-        if(version != null) {
-            Log.i(TAG, "Setting version download timestamp " + version + " to " + time);
-            if(time == null) {
-                this.editor.remove(version + DOWNLOADED_SUFFIX);
+    private void saveVersionInfo(final String version, final VersionInfo info) {
+        if(version != null && (info == null || !info.isBuiltin() && !info.isUnknown())) {
+            final VersionInfo update = info.setVersion(version);
+            Log.d(TAG, "Storing info for version [" + version + "] " + update.toString());
+            if(info == null) {
+                this.editor.remove(version + INFO_SUFFIX);
             } else {
-
-                final SimpleDateFormat sdf;
-                sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ");
-                sdf.setTimeZone(TimeZone.getTimeZone("CET"));
-                final String isoDate = sdf.format(time);
-
-                this.editor.putString(version + DOWNLOADED_SUFFIX, isoDate);
+                this.editor.putString(version + INFO_SUFFIX, update.toString());
             }
             this.editor.commit();
+        } else {
+            Log.d(TAG, "Not saving info for version: [" + version + "] " + info);
         }
     }
 
-    private String getVersionName(final String version) {
-        return this.prefs.getString(version + NAME_SUFFIX, "");
-    }
-
     public void setVersionName(final String version, final String name) {
         if(version != null) {
-            Log.i(TAG, "Setting version name " + version + " to " + name);
-            if(name == null) {
-                this.editor.remove(version + NAME_SUFFIX);
-            } else {
-                this.editor.putString(version + NAME_SUFFIX, name);
-            }
-            this.editor.commit();
+            Log.d(TAG, "Setting name for version [" + version + "] to " + name);
+            VersionInfo info = this.getVersionInfo(version);
```

Comment:
Note: VersionInfo should always be READ before we WRITE new data -- this will prevent overwriting any data from previous writes that were not synchronized to the Java code. Better safe than sorry.